### PR TITLE
clients/js: add viem compat wrapper

### PR DESCRIPTION
**Pull request by nhynes**

### CODE REVIEW

File: clients/js/src/compat/viem.ts [hunk 1]
```diff
@@ -33,18 +34,14 @@ export function wrapPublicClient<U extends PublicClient>(
...
-      async call(req) {
-        return upstream.call({
-          ...req,
-          data: await cipher.encryptEncode(req.data),
-        });
-      },
-    } as Hooks<U>,
-  );
+  return makeProxy(upstream, cipher, {
+    async call(req) {
```
Comment (aefhm) [inline clients/js/src/compat/viem.ts:38]: Noting that we `extend` publicClient strictly for signed queries.

File: clients/js/package.json [hunk 2]
```diff
@@ -47,12 +39,13 @@
     "@ethersproject/bytes": "^5.7.0",
     "@ethersproject/providers": "^5.7.1",
     "@ethersproject/rlp": "^5.7.0",
+    "@noble/hashes": "^1.3.1",
```
Comment (aefhm) [inline clients/js/package.json:42]: This is consolidated/merged.

File: clients/js/src/compat/viem.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  upstream.transport.request;
+  const cipher =
+    overrides?.cipher ??
+    lazyCipher(async () => {
+      const rtPubKey = await fetchRuntimePublicKey(transport, upstream.chain);
+      return X25519DeoxysII.ephemeral(rtPubKey);
+    });
+
+  return makeProxy<U>(upstream, cipher, {
+    async sendTransaction(req) {
```
Comment (aefhm) [inline clients/js/src/compat/viem.ts:68]: This will support Node better.

https://github.com/wevm/viem/discussions/1452

**Comment by BerkliumBirb** [comment]
Any progress on this one?

**Comment by aefhm** [comment]
> Any progress on this one?

@BerkliumBirb we will be releasing a new version pending https://github.com/oasisprotocol/sapphire-paratime/pull/303 soon

**Comment by aefhm** [comment]
Closing in favor of #303

**Comment by aefhm** [comment]
@BerkliumBirb we have released an alpha version [here](https://www.npmjs.com/package/@oasisprotocol/sapphire-viem-v2).

**Pull request closed without merging**